### PR TITLE
Verify No Metadata Pods Were Deployed

### PR DESCRIPTION
**Pull request by diegolovison**
```
erify No Metadata Pods Were Deployed :: Because the product inclu... | PASS |
------------------------------------------------------------------------------
Tests.Ods Dashboard.Data Science Pipelines.Data-Science-Pipelines-... | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Tests.Ods Dashboard.Data Science Pipelines                            | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Tests.Ods Dashboard                                                   | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Tests                                                                 | PASS |
1 test, 1 passed, 0 failed
==============================================================================

```

### CODE REVIEW

**Comment by bdattoma** [comment]
IMO, it would be better to move the checks added here inside one of the Pipeline UI tests we already have. In this way we would save execution time in Smoke suite.

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot [hunk 1]
```diff
@@ -66,6 +67,39 @@ Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Pag
     Delete Pipeline Server    ${PRJ_TITLE}
     Delete Data Science Project    ${PRJ_TITLE}
 
+Verify No Metadata Pods Were Deployed
+    [Documentation]    Because the product includes code that is currently deactivated, it is imperative to prevent any
```
Comment (jgarciao) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:71]: Could you rewrite the text according to the style guide? e.g. _Verifies that metadata pods are not created when running a data science pipeline, as this feature is currently disabled._

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot [hunk 2]
```diff
@@ -66,6 +67,39 @@ Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Pag
...
+    ...         occurrences of disabled elements from appearing unexpectedly.
+    [Tags]    Smoke
```
Comment (jgarciao) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:73]: Polarion ID missing

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot [hunk 3]
```diff
@@ -66,6 +67,39 @@ Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Pag
...
+    ...    press_cancel=${TRUE}
+    Open Data Science Project Details Page    ${PRJ_TITLE}
+    ${workflow_name}=    Create Pipeline Run    name=${PIPELINE_TEST_RUN_BASENAME}
+    ...    pipeline_name=${PIPELINE_TEST_NAME}    from_actions_menu=${FALSE}    run_type=Immediate
+    Open Data Science Project Details Page    ${PRJ_TITLE}
+    Wait Until Pipeline Last Run Is Started    pipeline_name=${PIPELINE_TEST_NAME}
+    ...    timeout=10s
+    Wait Until Pipeline Last Run Is Finished    pipeline_name=${PIPELINE_TEST_NAME}
+    ...    timeout=180s
+    @{pods} =    Oc Get    kind=Pod    namespace=${PRJ_TITLE}
```
Comment (jgarciao) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:96]: Lines 96-100 could be simplified using the existing keywords in ods-ci/ods_ci/tests/Resources/Page/OCPDashboard/Pods/Pods.robot like _Check If Pod Exists_

**Review by jgarciao — CHANGES_REQUESTED** [review]

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot (hunk 2 above)
Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:73]: No Polarion ID is required. This test will be removed as soon we have the metadata in the UI

Comment (diegolovison) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:73]: Uploading the result in Polarion and keeping it on track is enough.

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot [hunk 4]
```diff
@@ -66,6 +67,39 @@ Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Pag
...
+    FOR    ${pod}    IN    @{pods}
+        Log    ${pod['metadata']['name']}
+        Should Not Contain    ${pod['metadata']['name']}    ds-pipeline-metadata
+    END
+    Delete Data Science Project    ${PRJ_TITLE}
```
Comment (jgarciao) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:101]: I think it should be like this for this and the previous test:
`[Teardown]  Delete Data Science Project    ${PRJ_TITLE}`

If not, I think a failure in the first test will prevent the second one running properly

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot (hunk 2 above)
Comment (jgarciao) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:73]: In my opinion, you could create the test en Polarion and, once the metadata is enabled in the UI, modify the test (here and in Polarion) to check that the metadata is enabled.  It's not a big deal, so I leave it to your judgement.

**Review by jgarciao — APPROVED** [review]

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot [hunk 5]
```diff
@@ -66,6 +67,39 @@ Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Pag
     Delete Pipeline Server    ${PRJ_TITLE}
     Delete Data Science Project    ${PRJ_TITLE}
 
+Verify No Metadata Pods Were Deployed
```
[the remark this answers is not in the document]
Comment (jgarciao) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot]: Discussed with Diego offline

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot [hunk 6]
```diff
@@ -64,7 +65,40 @@ Verify User Can Create, Run and Delete A DS Pipeline From DS Project Details Pag
...
     Delete Pipeline    ${PIPELINE_TEST_NAME}
     Delete Pipeline Server    ${PRJ_TITLE}
-    Delete Data Science Project    ${PRJ_TITLE}
+    [Teardown]    Delete Data Science Project    ${PRJ_TITLE}
+
+Verify Pipeline Metadata Pods Are Not Deployed When Running Pipelines
+    [Documentation]    Verifies that metadata pods are not created when running a data science pipeline,
+    ...         as this feature is currently disabled.
+    [Tags]    Smoke
+    ...       Tier1
```
Comment (lugi0) [inline ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:74]: Isn't Tier1 == Sanity? 
Do you want this to be run in Smoke or Sanity?

**Review by lugi0 — APPROVED** [review]
LGTM but the Tags probably need a small change

File: ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot (hunk 6 above)
Comment (jgarciao) [reply ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/434__data-science-pipelines-ui.robot:74]: We have a lot of tests with Tier1 & Smoke tags, and others with Tier1 & Sanity. I think this is fine

**Pull request merged**